pkg/runtime/values/object.go

```diff
@@ -73,18 +76,29 @@ func (t *Object) Compare(other core.Value) int {
 
 		var res = 0
 
-		var val core.Value
-		var exists bool
+		sortedT := sort.StringSlice(t.Keys())
+		sortedT.Sort()
 
-		other.ForEach(func(otherVal core.Value, key string) bool {
-			res = -1
+		sortedOther := sort.StringSlice(other.Keys())
+		sortedOther.Sort()
+
+		var tVal, otherVal core.Value
+
+		for i := 0; i < len(t.value) && res == 0; i++ {
+			if sortedT[i] == sortedOther[i] {
+				tVal, _ = t.Get(NewString(sortedT[i]))
+				otherVal, _ = other.Get(NewString(sortedOther[i]))
+				res = tVal.Compare(otherVal)
```

Comment:
Let's compare the ``res``. If it does not equal to zero, we need to stop the iteration.

Reply:
Done